src/DiagramGenerator/Generator.php

```diff
@@ -47,7 +48,7 @@ public function buildDiagram(Config $config, $rootCacheDir, $boardTextureUrl, $p
     {
         $errors = $this->validator->validate($config);
         if (count($errors) > 0) {
-            throw new \Exception($errors->__toString());
+            throw new InvalidConfigException($errors->__toString());
```

Comment:
How will this change affect v2? The other code changes seem well encapsulated (unless the calling methods caught "ImagickException" and are now receiving a "CachedFileInvalidException"), but this change *only* affects the controller.